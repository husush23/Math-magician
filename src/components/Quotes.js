/*eslint-disable*/
import {useState, useEffect} from 'react';

function Quotes() {
  const [quoteData, setQuoteData] = useState(['Quote here']);
  const [error, setError] = useState(0);
  const [loading, setLoading] = useState(0);

  useEffect(() => {
    setLoading(1);
    const getData = async () => {
      try {
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=intelligence',
          {
            headers: {
              'X-Api-Key': 'l7joY6pxRAtosVpy+GGSVg==tL66pxRY0ioTzgkd',
            },
          }
        );

        if (!res.ok) {
          setError(1);
          return;
        }

        const data = await res.json();

        if (Array.isArray(data) && data.length > 0) {
          setLoading(0);
          setQuoteData(data[0]);
        } else {
          setError(1);
        }
      } catch (error) {
        setError(error);
      }
    };
    getData();
  }, []);

  if (error) return <h3>Something went wrong!</h3>;
  if (loading) return <h4>Loading....</h4>;

  return (
    <div className='qoute-cont'>
      <p>{quoteData && quoteData.quote}</p>
      <h2 className='author'>~ {quoteData.author}</h2>
    </div>
  );
}

export default Quotes;
