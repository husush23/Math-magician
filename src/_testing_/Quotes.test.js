import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

test('renders correctly', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
