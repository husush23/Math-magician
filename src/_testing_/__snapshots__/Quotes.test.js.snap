// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="qoute-cont"
>
  <p />
  <h2
    className="author"
  >
    ~
  </h2>
</div>
`;
